rfd/0191-workload-id-config-ux.md

```diff
@@ -0,0 +1,1454 @@
+  // SSO.
+  Traits traits = 4;
+}
+```
+
+Example (marshalled as YAML):
+
+```yaml
+name: bot-gitlab-workload-identity
+is_bot: true
+bot_name: gitlab-workload-identity
+traits:
+- name: foo
+  values:
+  - bar
+  - bizz
+- name: logins
+  values:
+  - root
+```
+
+#### Workload Attributes
+
+The workload attributes are provided by the `tbot` agent at the time of
+attestation as part of the RPC request.
+
+Depending on whether workload attestation has been performed, or which form of
+attestation has been performed, the attributes available will differ.
+
+```protobuf
+package teleport.workloadidentity.v1;
+
+// Attributes pertaining to unix workload attestation.
+message WorkloadUnix {
+  // Whether or not this attestation method has been performed successfully.
+  bool attested = 1;
+  // The PID of the process that connected to the workload API.
+  int32 pid = 2;
+  // The UID of the process that connected to the workload API.
+  uint32 uid = 3;
+  // The GID of the process that connected to the workload API.
+  uint32 gid = 4;
+  // -- snipped for conciseness --
+}
+
+// Attributes pertaining to Kubernetes workload attestation.
+message WorkloadKubernetes {
+  // Whether or not this attestation method has been performed successfully.
+  bool attested = 1;
+  // The name of the pod that connected to the workload API.
+  string pod_name = 2;
+  // The namespace of the pod that connected to the workload API.
+  string namespace = 3;
+  // The service account of the pod that connected to the workload API.
+  string service_account = 4;
+  // -- snipped for conciseness --
+}
+
+// Attributes sourced from the workload attestation process.
+message WorkloadAttributes {
+  // Attributes pertaining to Unix workload attestation.
+  WorkloadUnix unix = 1;
+  // Attributes pertaining to Kubernetes workload attestation.
+  WorkloadKubernetes k8s = 2;
+}
+```
+
+Example (marshalled as YAML):
+
+```yaml
+unix:
+  attested: true
+  pid: 1234
+  uid: 1000
+  gid: 1000
+k8s:
+  attested: true
+  pod_name: my-pod-ab837c
+  namespace: my-namespace
+  service_account: my-service-account
+```
+
+#### Join Attributes
+
+The `join` attributes are sourced from the attestation process that occurs when
+a principal (Bot, Agent etc) initially authenticates using a Join Token.
+
+At the top level of the join attributes, the `meta` key will hold attributes
+that apply to all joins. For example, the name of the join token used and the
+method of the join.
+
+A top level key will exist for each join method that is supported by Teleport.
+Under this key, attributes specific to that join method will be present.
+Typically, these attributes will be sourced directly from some identity
+document provided by the principal upon joining (e.g an ID Token issued by
+GitLab).
+
+Specification:
+
+```protobuf
+package teleport.workloadidentity.v1;
+
+// Attributes present for all joins, including metadata about the join token
```

Comment:
Would further join methods like GitHub get added to the roadmap or wait for customer request?

Reply:
I'll be adding this for all our existing join methods out of the box, this is just more concise to avoid turning the RFD into a massive wall of protobuf.